layers/arp.go

```diff
@@ -39,6 +40,10 @@ func (arp *ARP) LayerType() gopacket.LayerType { return LayerTypeARP }
 
 // DecodeFromBytes decodes the given bytes into this layer.
 func (arp *ARP) DecodeFromBytes(data []byte, df gopacket.DecodeFeedback) error {
+	if len(data) < 8 || len(data) < int(8 + 2*data[4] + 2*data[5]) {
```

Comment:
Can you change this to first check (data) < 8 and then later check with `HwAddressSize` and `ProtAddressSize` after they have been decoded? This way it is more clear what is going on